Update release process

laanwj opened the pull request:
The actual release process had quite diverged from what was written here, also clarify things a bit and update the style to be like our other .md files. More like "rewrite the entire thing".

Best to read the rendered version: https://github.com/laanwj/bitcoin/blob/2016_04_update_release_process/doc/release-process.md

Closes #7595 as well by removing the hardcoded version number from `verify.sh`.

fanquake commented on doc/release-process.md:
````diff
@@ -111,71 +124,88 @@ The gbuild invocations below <b>DO NOT DO THIS</b> by default.
+```
+
+The list of files should be:
+```
````
Should we use {VERSION} here in place of hard coded numbers

laanwj replied:
Sure!

Commit:
[contrib] Remove reference to sf and add doc to verify.sh

MarcoFalke commented:
Concept ACK. Confirmed changes to verify.sh are working

laanwj commented:
Fixed @fanquake 's nit, cherry-picked #7882 into this.

laanwj committed:
doc: Update release process

The actual release process quite diverged from what was written here,
also clarify things a bit.

laanwj committed:
contrib: remove hardcoded version from verify.sh

Closes #7595 as by removing the hardcoded version number from
`verify.sh`.

MarcoFalke commented on doc/release-process.md:
```diff
@@ -111,71 +125,86 @@ The gbuild invocations below <b>DO NOT DO THIS</b> by default.
-Commit your signature for the signed OS X/Windows binaries:
+Wait for Windows/OS X detached signatures:
+
+    Once the Windows/OS X builds each have 3 matching signatures, they will be signed with their respective release keys.
```
You can't add a new line here and keep the pre-whitespace. This will mess with the formatting

laanwj replied:
This should probably be a - list

PRabahy commented on doc/release-process.md:
```diff
@@ -111,71 +125,86 @@ The gbuild invocations below <b>DO NOT DO THIS</b> by default.
+    git add ${VERSION}-linux/${SIGNER}
+    git add ${VERSION}-win-unsigned/${SIGNER}
+    git add ${VERSION}-osx-unsigned/${SIGNER}
+    git commit -a
```
I always end up doing something like
git commit -a -S -m "${SIGNER}'s unsigned gitian build for ${VERSION}"

The extra -S is probably unnecessary because we end signing a commit containing signed files, but I don't think it should hurt anything and GPG already has to be set up correctly for gsign to work.

MarcoFalke replied:
You'd need to `git config ...` for `git commit -S` to work

laanwj replied:
There's not really a need to sign commits with signatures in them, I think that'll just result in confusion which signatures are used.

MarcoFalke commented on doc/release-process.md:
```diff
@@ -111,71 +125,86 @@ The gbuild invocations below <b>DO NOT DO THIS</b> by default.
-	./bin/gverify -v -d ../gitian.sigs/ -r ${VERSION}-win-unsigned ../bitcoin/contrib/gitian-descriptors/gitian-win.yml
-	./bin/gverify -v -d ../gitian.sigs/ -r ${VERSION}-osx-unsigned ../bitcoin/contrib/gitian-descriptors/gitian-osx.yml
-
-	popd
```
Are you sure this is no longer required here?

laanwj replied:
The verifies? They're still there, this is just a formatting change

MarcoFalke replied:
No, the popd

laanwj replied:
Well if there should be a popd it should at least be balanced by a pushd. Out of the blue it is very strange.

MarcoFalke replied:
You have moved it up instead. See my other comment above ;)

MarcoFalke replied:
I think they were "balanced" before. (Not necessarily in each section, though)

laanwj replied:
Hah, okay, having pushd/popd open over sections is weird, I must have not been the only one to never get that point. I'll add a pushd and popd around this.

MarcoFalke commented on doc/release-process.md:
```diff
@@ -1,83 +1,97 @@
-###Fetch and create inputs: (first time, or when dependency versions change)
+    pushd ./gitian-builder
+    git pull
+    popd
```
If you add this, you'd need to add `      pushd ./gitian-builder ` at the other places?

laanwj replied:
Yes using pushd/popd consistently everywhere would make sense.

MarcoFalke commented:
utACK cdb7804

fanquake commented:
utACK cdb7804

laanwj commented:
squashed 2873dbd 3707e43 373bfef 6e25333 cdb7804 into c907f4d 182bec4 f154470
